### PR TITLE
[alphonse] Round-2: Compose RFF + compile-fix (single-delta composition test)

### PR DESCRIPTION
**Pull request by morganmcg1**
# Round-2 Assignment: Compose RFF + compile-fix (single delta)

## Hypothesis

Two proven wins now on the tay board:
- **PR #33 (fern):** RFF coord features alone → test_abupt 17.77 (−10.3% vs PR #30)
- **PR #40 (alphonse):** Compile fix alone → test_abupt 17.25 (−2.9% vs PR #33)

Both were measured independently on different baselines. This PR tests whether they **compose** by adding RFF on top of the new compiled baseline from PR #40. If additive, projected outcome is ~15.5 abupt — which would beat yi's 15.82 reference.

**Single delta:** PR #40 config + `--rff-num-features 32 --rff-sigma 1.0`. No other changes.

### Run the experiment

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent alphonse \
  --wandb-name "alphonse/round2-rff-compiled-512d" \
  --wandb-group "tay-round2-rff-compiled" \
  --volume-loss-weight 2.0 \
  --batch-size 4 \
  --validation-every 1 \
  --lr 5e-5 --weight-decay 5e-4 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.9995 \
  --rff-num-features 32 --rff-sigma 1.0 \
  --gradient-log-every 100 --weight-log-every 100 \
  --no-log-gradient-histograms
```

**No `--no-compile-model`** — the compile bug was fixed in PR #40. Compiled training is the default now.

### What to report

After training completes, post a comment with **all** `test_primary/*` metrics from the best-val checkpoint:

| Metric | This run | PR #40 (baseline) | PR #33 |
|---|---:|---:|---:|
| `abupt` | ? | 17.25 | 17.77 |
| `surface_pressure` | ? | 10.92 | 11.20 |
| `wall_shear` | ? | 18.33 | 18.68 |
| `volume_pressure` | ? | 14.71 | 16.13 |
| `tau_x` | ? | 15.73 | 16.20 |
| `tau_y` | ? | 21.80 | 21.81 |
| `tau_z` | ? | 23.07 | 23.54 |

Also report:
- W&B rank-0 run ID
- Wall-clock time and number of epochs reached
- Best val_abupt

### Expected outcome

If RFF and compile compose additively, projected test_abupt is ~15.5 (below yi's 15.82). The val curve at PR #33 had val 17.06 with RFF/no-compile; the same mechanism on compiled runs should reach ~15.5–16 val and corresponding test.

## Current baseline to beat

| Metric | tay best (PR #40) | yi best | AB-UPT |
|---|---:|---:|---:|
| `test_primary/abupt_axis_mean_rel_l2_pct` | **17.25** | 15.82 | — |
| `test_primary/surface_pressure_rel_l2_pct` | **10.92** | 9.99 | 3.82 |
| `test_primary/wall_shear_rel_l2_pct` | **18.33** | 16.60 | 7.29 |
| `test_primary/volume_pressure_rel_l2_pct` | **14.71** | 14.21 | 6.08 |
| `test_primary/wall_shear_x_rel_l2_pct` | **15.73** | 14.27 | 5.35 |
| `test_primary/wall_shear_y_rel_l2_pct` | **21.80** | 19.49 | 3.65 |
| `test_primary/wall_shear_z_rel_l2_pct` | **23.07** | 21.12 | 3.63 |

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR (auto-posted, alphonse pod stalled post-train):

This run finished cleanly in W&B but the pod did not run the report-results step. Posting verified W&B metrics here so we can merge.

## Results — NEW SOTA: test_abupt 14.550

W&B run `28l4yanr` (rank 0), group `tay-round2-rff-compiled`. State: finished. Runtime: 284.9 min. Best epoch (val): 16 of 50. Selection metric: `val_primary/abupt_axis_mean_rel_l2_pct` from EMA. No NaN/Inf, no error keys.

| Metric | PR #46 (NEW SOTA) | PR #39 Lion (prev SOTA) | Δ abs | Δ % |
|---|---:|---:|---:|---:|
| `test_primary/abupt_axis_mean_rel_l2_pct` | **14.550** | 15.43 | **−0.88** | **−5.7%** |
| `surface_pressure` | 8.628 | 9.45 | −0.82 | −8.7% |
| `wall_shear` | 14.882 | 16.28 | −1.40 | −8.6% |
| `volume_pressure` | 15.032 | 13.83 | +1.20 | +8.7% |
| `tau_x` | 12.901 | 13.91 | −1.01 | −7.3% |
| `tau_y` | 17.281 | 19.58 | −2.30 | −11.7% |
| `tau_z` | 18.907 | 20.40 | −1.49 | −7.3% |

Best val_primary/abupt at epoch 16: **13.487** (val→test gap ~+1.06).

### Hyperparameters

`optimizer=AdamW (default), lr=5e-5, weight_decay=5e-4, ema_decay=0.9995, rff_num_features=32, rff_sigma=1.0, compile_model=True, model=4L/512d/8h/128 slices, batch=4, ema_decay=0.9995`.

## Mechanism

This is the AdamW + RFF + compile cell of the 2×2×2 factorial (PR #40 + RFF). Compile lets training reach epoch 16 (vs ~9 uncompiled for PR #33; vs 12 the round-1b baseline target was set for). The deeper training surfaces the headroom in the **wall-shear axes specifically**: tau_y −11.7%, wall_shear (mag) −8.6%, surface_pressure −8.7% all move materially. tau_y leads, which is consistent with the spectral-bias hypothesis (RFF + more epochs to saturate the high-frequency components).

The one regression: `volume_pressure` +8.7%. Same hyperparameter (`volume_loss_weight=2.0`) as PR #39, so this likely reflects the AdamW vs Lion difference for volume — Lion's sign-update gives more uniform per-channel gradient signal. **Lion + RFF + compile (PR #52 tanjiro, queued) should fix the volume regression and stack with the surface/wall_shear gains here.**

## Implications for queued PRs

- **PR #50 nezuko (Lion + compile, no RFF)**: queued. Expected to land between PR #39 (Lion-only, 15.43) and this PR (RFF + compile, 14.55). The compile lever alone is worth ~−5% (PR #46 vs #33: 14.55 vs 17.77). On Lion base, compile should give ~14.5–15.0 abupt.
- **PR #51 edward (Lion + RFF, no compile)**: queued. RFF was −10% vs PR #30 baseline; on Lion base, expected ~13.5–14.0.
- **PR #52 tanjiro (Lion + RFF + compile)**: queued. The full triple — if all three levers are independent, expected ~12.5–13.5 abupt. **This is the headline run.**
- **PR #54 fern (Lion + per-axis tau_y/tau_z weights)**: just assigned. Orthogonal to the optimizer/feature/compile axis — addresses the binding 5×+ gap on tau_y (3.65 AB-UPT) and tau_z (3.63). Stacks with all of the above.

Marking ready for review and merging.
